client/me/application-passwords/index.jsx

```diff
@@ -3,106 +3,85 @@
 /**
  * External dependencies
  */
-
-import React from 'react';
-import createReactClass from 'create-react-class';
-import { localize } from 'i18n-calypso';
-import debugFactory from 'debug';
-const debug = debugFactory( 'calypso:application-passwords' );
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
 import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-/* eslint-disable no-restricted-imports */
-import observe from 'lib/mixins/data-observe';
-/* eslint-enable no-restricted-imports */
 import AppPasswordItem from 'me/application-password-item';
-import notices from 'notices';
-import SectionHeader from 'components/section-header';
 import Button from 'components/button';
-import Gridicon from 'gridicons';
-import FormFieldset from 'components/forms/form-fieldset';
-import FormTextInput from 'components/forms/form-text-input';
-import FormLabel from 'components/forms/form-label';
+import Card from 'components/card';
 import FormButton from 'components/forms/form-button';
 import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
 import FormSectionHeading from 'components/forms/form-section-heading';
-import Card from 'components/card';
-import classNames from 'classnames';
-import { errorNotice } from 'state/notices/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import QueryApplicationPasswords from 'components/data/query-application-passwords';
+import SectionHeader from 'components/section-header';
+import {
+	clearNewApplicationPassword,
+	createApplicationPassword,
+} from 'state/application-passwords/actions';
+import { getApplicationPasswords, getNewApplicationPassword } from 'state/selectors';
 import { recordGoogleEvent } from 'state/analytics/actions';
 
-const ApplicationPasswords = createReactClass( {
-	displayName: 'ApplicationPasswords',
-	mixins: [ observe( 'appPasswordsData' ) ],
-
-	componentDidMount: function() {
-		debug( this.displayName + ' React component is mounted.' );
-	},
+class ApplicationPasswords extends Component {
+	initialState = {
```

Comment:
This is fine now, but carelessness can make for subtle bugs if this were to be mutated. How do you feel about something like this?

https://github.com/Automattic/wp-calypso/blob/4c28909429d8cbede85a76bdf50e5511620ca1ac/client/jetpack-connect/signup.js#L58-L68

Reply:
👍 Freezing and making it static done in https://github.com/Automattic/wp-calypso/pull/23926/commits/385fa0ef59e646a1b54cf84ff4aec8bd8fb2e839

I've passed on creating a separate method, as it will be used just once at this point.